### ISSUE TITLE
Use of Assembly.Location makes this incompatible with single file EXEs

### ISSUE THREAD

**Issue by someguy20336** [daveaglick/Buildalyzer#214]
For a project I am working on, I was interested in trying out the [single file EXE](https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview?tabs=cli).  Unfortunately, due to the use of `Assembly.Location` [in at least one spot](https://github.com/daveaglick/Buildalyzer/blob/main/src/Buildalyzer/ProjectAnalyzer.cs#L255) (which is [incompatible with single file EXEs](https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview?tabs=cli#api-incompatibility)) causes the application to hang indefinitely while attempting to build the project out of process.

The output of the [`ProcessRunner`](https://github.com/daveaglick/Buildalyzer/blob/main/src/Buildalyzer/ProjectAnalyzer.cs#L167) shows the logger switch causes the build to fail, and apparently never return:

![image](https://user-images.githubusercontent.com/38168572/190872350-f54c5241-5cf2-4ce8-965f-42eedd7159f5.png)

Of course, the workaround is to _not_ do a single file EXE, but the single file feature is quite nice.

**Comment by someguy20336** [comment]
I haven't worked with the single file EXE too much yet, but I wonder if there is a way to access the bundled (?) assembly via some "path".  Or alternatively, extract a single bundled assembly. So far, no luck with just some simple Google searches.  I haven't tested anything out yet, though.

**Comment by daveaglick** [comment]
Unfortunately, assembly loading can be tricky and the location of the logging assembly could be in a few places depending on platform, delivery, etc. That's why `Assembly.Location` works well here - it lets the code completely ignore how the logging assembly is linked and from where and relies on the runtime to do that and supply the information. It's _probably_ possible to embed the logging assembly directly in the Buildalyzer NuGet kind of like an unmanaged library instead of relying on a transitive dependency to bring it in, and then we might be able to find it using NuGet APIs or some other mechanism.

All of that sounds like a non-trival amount of work and testing for a feature that's non-critical though. It's not that I'm opposed to the idea, it's just that priority-wise it falls pretty low on a list that I'm already struggling (and mostly failing) to keep up with.

TL;DR: I'd probably accept a PR as long as I had confidence it wouldn't create more problems down the road or on other platforms, but I'm unlikely to work on this myself.

**Comment by someguy20336** [comment]
I wouldn't be offended if you just want to close this issue. I spent a little time trying to find a good way to do this and it just seems overly complicated for the simple workaround of "don't deploy as single file". Maybe some future advancements in single file deployments would make worth revisiting later.

**Comment by daveaglick** [comment]
I'll close the issue for now given I'm unlikely to have the bandwidth for much further exploration, but if you (or anyone else) ends up with new information or a workaround, I'd be happy to reopen it.
